reactor-core/src/test/java/reactor/core/publisher/MonoIgnoreThenTest.java

```diff
@@ -1,29 +1,149 @@
+/*
+ * Copyright (c) 2011-2018 Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package reactor.core.publisher;
 
+import java.time.Duration;
 
+import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 import org.reactivestreams.Publisher;
-import org.reactivestreams.Subscription;
+
+import reactor.core.Disposable;
 import reactor.core.Scannable;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+import reactor.test.StepVerifier;
+import reactor.test.publisher.TestPublisher;
 import reactor.test.subscriber.AssertSubscriber;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-public class MonoIgnoreThenTest {
+class MonoIgnoreThenTest {
+
+	@Nested
+	class ThenIgnoreVariant {
+
+		Publisher<Void> scenario(){
```

Comment:
Minor, but it seems it can be inlined since the method is used in one case only

Reply:
it is actually a good polish of the base PR. let me just do that and the license header that we've noticed, and forward merge that separately (as it will then be trivial), if that sounds good?

Reply:
done in 64649ff5171bac1cf385f63b8d5f9f6749eda597